### ISSUE TITLE
RuneLite Launched via Jagex Launcher on MacOS (Monterey) M1 Throws 400 Error

### ISSUE THREAD

**Issue by clrosier** [runelite/runelite#18133]
### Type

Incorrect behavior

### Operating System

MacOS

### OS Version

MacOS Monterey

### Bug description

Upon attempting to login via Runelite, I am met with the error:
`Failed to login. Please try again.`

This is my work laptop (with clearance to have runescape installed).  It runs an outbound filter software called "Netskope".
I am running the client with flags:

1. Debug mode (for extra verbosity in logging)
2. Skipping tls verification (for ssl issues I get without it checked)

Additionally, if I launch the Official Client via the Jagex Launcher, I can login successfully.

### Screenshots or videos

![image](https://github.com/user-attachments/assets/ba622602-8a7e-4b9b-b71d-0d5717f2ca98)

### RuneLite version

RuneLite version: 2.7.1
Launcher Version: 1.0.0

### Logs

[client.log](https://github.com/user-attachments/files/16515167/client.log)
[launcher.log](https://github.com/user-attachments/files/16515168/launcher.log)

**Comment by clrosier** [comment]
I've tried to sift through the RuneLite codebase, but I am unable to find the exact spot that would generate the blank 400 error seen in the logs.

**Comment by cdfisher** [comment]
As was mentioned on Discord, this seems to be a problem with your Macbook's outbound filtering and the Jagex Launcher rather than a RuneLite issue.

**Comment by cdfisher** [comment]
> I do suspect this as well, but it would be helpful if the location that was returning this error could be pointed out, that way a root cause could be definitively documented. If you have any hint at where that might be, I could open a PR to add the verbosity

As Adam told you this morning, it comes from the Jagex Launcher, not from RuneLite, so there's not anything to PR.

**Comment by clrosier** [comment]
Got it.  So then is it safe to make the assumption that RuneLite is logging a blank response body because the Jagex Launcher is providing a blank response to the RuneLite client?

**Comment by Adam-** [comment]
RuneLite is not really even logging this, we just have the Jagex client logs piped into the RuneLite logs, since otherwise it goes to stdout. Is it possible for you to tcpdump/wireshark the request and see what it is doing?

**Comment by clrosier** [comment]
I can attempt to do a network capture on it, WireShark will probably send a few alarm bells off to the Security team at work so I'll go the safer route :)

**Comment by goshcharlotte** [comment]
I realise this is closed now but I've been having exactly the same issue on macOS Tahoe 26.5.2, RL version 2.7.1. I tried to open a support ticket with Jagex but since the official osrs client is working fine and _technically_ the issue is happening in RuneLite, they said they can't do anything to help and closed my ticket.

I get this in my client logs:
```
2026-08-12 09:09:47 BST [Client] DEBUG injected-client - Game state changed: LOGGING_IN (state: 20, 2 step: false)
2026-08-12 09:09:47 BST [Client] ERROR injected-client - Client error: Login authentication error. Response code: 400 Error servicing REST query: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target Response body:
2026-08-12 09:09:47 BST [Client] DEBUG injected-client - Game state changed: LOGIN_SCREEN (state: 10, 2 step: false)
```

I'm not able to turn off my outbound Netskope filtering, but I wondered if anyone else has had this issue and managed to resolve it in the past 2 years?

**Comment by goshcharlotte** [comment]
WAIT literally 10 minutes later and I've fixed it! I'm definitely not an expert on how certificates work but I believe Netskope intercepts certificates for all your traffic and adds its own netskope root cert(?), and when Netskope is first installed it also installs its cert on your machine - but RuneLite isn't checking your machine, it's checking the `cacerts` within the embedded JRE, which doesn't have the netskope cert.

So, in the directory `/Applications/RuneLite.app/Contents/Resources/jre/bin/` I found the `keytool` util, and was able to import the netskope cert into the JRE.

Here are the steps I took to get it working:

1. Visited `account.runescape.com` (or basically anything tbh) and inspected the certificate, clicked the root netskope cert and downloaded the pem file
2. Following [this answer](https://stackoverflow.com/a/13992135) I ran `openssl x509 -outform der -in ~/Downloads/account-runescape-com.pem -out ~/Downloads/account-runescape.der`
3. From the directory `/Applications/RuneLite.app/Contents/Resources/jre/bin/` I ran `./keytool -import -alias netskope -keystore ../lib/security/cacerts -file ~/Downloads/account-runescape.der`

And now I'm able to log in via the Jagex launcher, and have even been able to re-enable TLS verification in the RuneLite config!
